sigstore/verify/models.py

```diff
@@ -240,19 +259,34 @@ def from_bundle(
             )
         tlog_entry = tlog_entries[0]
 
-        inclusion_proof = LogInclusionProof(
-            log_index=tlog_entry.inclusion_proof.log_index,
-            root_hash=tlog_entry.inclusion_proof.root_hash.hex(),
-            tree_size=tlog_entry.inclusion_proof.tree_size,
-            hashes=[h.hex() for h in tlog_entry.inclusion_proof.hashes],
-        )
+        # NOTE: We only load the inclusion proof from the bundle if we know that
```

Comment:
Nit, should this note say for both online and offline verification?

Reply:
Yep -- we always retrieve the inclusion proof from an online Rekor instance unless the user has explicitly told us to prefer offline sources with `--offline`.

We did that because it was the only way to get the inclusion proof before we had bundles, but maybe that's no longer necessary -- is there client guidance around when clients should prefer online lookups versus in-situ bundle materials?

CC @znewman01

Reply:
Both online and offline give you the same guarantees, trusting what comes from Rekor. Online has a few negatives: Reliability (since Rekor must be online), and privacy (revealing what entries you care about to Rekor). I'd say to prefer the bundle when presented.

Long term, there's a few options for stronger guarantees of inclusion:
* Online querying of distributors (gatherers of witness statements) or witnesses themselves, but this comes with some privacy tradeoffs
* Offline bundles containing a set of witnessed checkpoints

The first is just changing client behavior, the second would be a bundle spec update.

In terms of the client, I'd prefer offline over online, mostly from a performance/reliability standpoint.

Reply:
Makes sense, thank you! I'll tweak to reflect that.

Reply:
Sorry, one last question (since I'm getting myself twisted up with online/offline again): isn't an inclusion proof retrieved from the bundle no stronger than an offline SET verification?

In other words: an adversary who can issue valid-looking SETs can also produce an valid-looking (from an offline context) inclusion proofs, including signed checkpoints on those inclusion proofs. 

So, in an "offline" context, is there any value to doing anything other than the SET verification? Correspondingly, in an "online" context, doesn't the inclusion proof + checkpoint _need_ to be retrieved online to ensure that it hasn't been forged from a split view of the log?

Reply:
![checkpoint-verify-kanon](https://user-images.githubusercontent.com/8418760/235751751-99f7d257-c996-4060-be68-b2beb02c0d63.png)

![checkpoint-verify-offline](https://user-images.githubusercontent.com/8418760/235751749-f75e1c69-f548-4652-9ff4-6eca8cd80a78.png)

Reply:
Makes sense, thank you! I think that ties a bow around it for me; I can refactor this to do the following:

1. Offline mode: require SET, opportunistically verify the inclusion proof and checkpoint if present
2. Online mode: require SET, require inclusion proof and checkpoint (retrieving them from Rekor if not present in the bundle)